### PR TITLE
crud assignment

### PR DESCRIPTION
**Pull request by poudelprakash**

### CODE REVIEW

**Comment by grishmashrestha** [comment]
@CuriousSugam :+1: Please include comments regarding the pull request like the goal for the task and how you achieved it in the pull request so that it is easier for the reviewer to review.

**Review by grishmashrestha — APPROVED** [review]

**Review by ojash — APPROVED** [review]

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  def index
+    @articles = Article.all
+  end
+
+  # this is executed when url pattern is /articles/  & method: POST
+  def create
+    # render plain: params[:article].inspect
+    @article = Article.new(article_params)
+
+    @article.save
```
Comment (poudelprakash) [inline app/controllers/articles_controller.rb]: If save fails? why not have if else condition just like you have implemented in update action?

Comment (poudelprakash) [reply app/controllers/articles_controller.rb]: maybe you would also like to notify users on if actions were successful, how about we start implementing flash?
http://api.rubyonrails.org/classes/ActionDispatch/Flash.html

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    if @article.update(article_params)
+      redirect_to @article
+    else
+      render 'edit'
+    end
+  end
+
+  # this is executed when url pattern is /articles/{id} & method: GET
+  def show
+    @article = Article.find(params[:id])
```
Comment (poudelprakash) [inline app/controllers/articles_controller.rb]: `@article = Article.find(params[:id])` is repeated in `update`, `show` and `destroy`.... such repeating code fit better in separate function, what do you say @CuriousSugam ?

**Comment by poudelprakash** [comment]
@CuriousSugam Please put .idea/ in .gitignore

File: config/environments/development.rb~ [hunk 3]
```diff
@@ -0,0 +1,54 @@
+Rails.application.configure do
```
Comment (poudelprakash) [inline config/environments/development.rb~:1]: this temp file should not be committed, let's add `*~` in `.gitignore`

File: config/locales/devise.en.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# Additional translations at https://github.com/plataformatec/devise/wiki/I18n
```
Comment (poudelprakash) [inline config/locales/devise.en.yml:1]: Journey of Internationalization starts here...You may want to read more about it:
http://guides.rubyonrails.org/i18n.html

File: config/routes.rb [hunk 5]
```diff
@@ -1,3 +1,9 @@
 Rails.application.routes.draw do
+  devise_for :models
```
Comment (poudelprakash) [inline config/routes.rb:2]: `models`? or should it be name of model instead?

File: app/models/model.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class Model < ApplicationRecord
```
Comment (poudelprakash) [inline app/models/model.rb:1]: You actually created a model named model! I am bit worried now...

Time to read https://github.com/plataformatec/devise#getting-started again

**Pull request merged**